fenix/app/src/androidTest/java/org/mozilla/fenix/ui/SettingsBasicsTest.kt

```diff
@@ -167,4 +167,25 @@ class SettingsBasicsTest {
             }
         }
     }
+
+    @SmokeTest
+    @Test
+    fun verifyHomepageOptionSummaryUpdatesTest() {
+        homeScreen {
+        }.openThreeDotMenu {
+        }.openSettings {
+            verifyHomepageButtonSummary("Open on homepage after four hours")
+        }.openHomepageSubMenu {
+            verifySelectedOpeningScreenOption("Homepage after four hours of inactivity")
+            clickOpeningScreenOption("Homepage")
+            verifySelectedOpeningScreenOption("Homepage")
+        }.goBack {
+            verifyHomepageButtonSummary("Open on homepage")
+        }.openHomepageSubMenu {
+            clickOpeningScreenOption("Last tab")
+            verifySelectedOpeningScreenOption("Last tab")
+        }.goBack {
+            verifyHomepageButtonSummary("Open on last tab")
```

Comment:
Can we use the string resources?

https://github.com/mozilla-mobile/firefox-android/blob/32bb398127bacccb268a272091d2e62b8d72d6b9/fenix/app/src/main/java/org/mozilla/fenix/settings/SettingsFragment.kt#L604

Reply:
Hi @AaronMT ,

We don't use string resources in the test class. (never seen a test using a string resource)

We have quite a lot of tests written in a similar fashion, verify the hard coded string.

Maybe @sv-ohorvath can also weigh in on this, but I've had this discussion with @Mugurell a while ago, and it's not a really good practice to rely only on the string resources.
Someone can make a mistake by setting a wrong string to a resource.
If that happens the test will never fail, and will not catch in time a real bug in our app.